File: src/track.cpp
#include "track.h"
#include "musiclib.h"

Track::Track() {
}

Track::Track(const QJsonObject& json) {
    fromJson(json);
}

void Track::fromJson(const QJsonObject& json) {
    album_ = json.value("album").toString();
    artist_ = json.value("artist").toString();
    genre_ = json.value("genre").toString();
    comment_ = json.value("comment").toString();
    track_ = json.value("track").toInt();
    title_ = json.value("title").toString();
    mrl_ = json.value("mrl").toString();
    path_ = json.value("path").toString();
    length_ = json.value("length").toInt();
    lengthString_ = json.value("lengthString").toString();
    year_ = json.value("year").toInt();
    disc_ = json.value("discNumber").toInt();
    id_ = json.value("id").toInt();
}

QJsonObject Track::toJson() const {
    QJsonObject tmp;

    tmp.insert("album", album_);
    tmp.insert("artist", artist_);
    tmp.insert("genre", genre_);
    tmp.insert("comment", comment_);
    tmp.insert("track", track_);
    tmp.insert("title", title_);
    tmp.insert("mrl", mrl_);
    tmp.insert("path", path_);
    tmp.insert("length", length_);
    tmp.insert("lengthString", lengthString_);
    tmp.insert("year", year_);
    tmp.insert("discNumber", disc_);
    tmp.insert("id", id_);

    return tmp;
}

File: PlaylistProvider/Track.cs
using System;
using Newtonsoft.Json;

namespace PlaylistProvider
{
    public class Track
    {
        public string album = "";
        public string artist = "";
        public string genre = "";
        public string comment = "";
        public string track = "";
        public string title = "";
        public string mrl = "";
        public string path = "";
        public int length = 0;
        public string lengthString = "";
        public int year = 0;
        public int discNumber = 0;
        public int id = 0;

        public static Track FromJson(string json)
        {
            return JsonConvert.DeserializeObject<Track>(json);
        }

        public string ToJson()
        {
            string output = JsonConvert.SerializeObject(this);
            return output;
        }
    }
}
